### PR TITLE
Update the README to give a basic description of the Council and the purpose of the repo

### PR DESCRIPTION
**Pull request by rylev**
While I have quite a few ideas of what would be useful to have on the README, I'd like to start with the basics.

To get consensus on this PR, I'd like to borrow the process we used in the governance RFC working group which is just checkboxes in the PR description. We'll probably want to improve this process over time including potentially using something like RFC bot. 

Checkboxes: 
- [x] @carols10cents 
- [x] @eholk 
- [x] @ehuss 
- [x] @jackh726 
- [x] @khionu 
- [x] @Mark-Simulacrum 
- [x] @m-ou-se 
- [x] @rylev 
- [x] @theJPster

### CODE REVIEW

**Comment by m-ou-se** [comment]
It's probably useful if this readme has a link to our #council Zulip channel and [team page on the website](https://www.rust-lang.org/governance/teams/leadership-council).

File: README.md [hunk 1]
```diff
@@ -1,2 +1,16 @@
-# leadership-council
-Home of the Rust Leadership Council
+# The Rust Project Leadership Council
+
+This repo is the home for documentation related to the Leadership Council for the Rust Programming Language Project (a.k.a. the Council).
+
+The Council was established by [RFC #3392][rfc]. The RFC describes the Council at a high-level as the following:
+
+> This RFC establishes a Leadership Council as the successor of the core team and the new governance structure through which Rust Project members collectively confer the authority to ensure successful operation of the Project. The Leadership Council delegates much of this authority to teams (which includes subteams, working groups, etc.3) who autonomously make decisions concerning their purviews. However, the Council retains some decision-making authority, outlined and delimited by this RFC.
```
Comment (jackh726) [inline README.md:7]:
```suggestion
> This RFC establishes a Leadership Council as the successor of the core team and the new governance structure through which Rust Project members collectively confer the authority to ensure successful operation of the Project. The Leadership Council delegates much of this authority to teams (which includes subteams, working groups, etc.) who autonomously make decisions concerning their purviews. However, the Council retains some decision-making authority, outlined and delimited by this RFC.
```
Typo

File: README.md [hunk 2]
```diff
@@ -1,2 +1,16 @@
...
+>
+> The Council will be composed of representatives delegated to the Council from each top-level team.
```
Comment (eholk) [inline README.md:9]:
```suggestion
> The Council is composed of representatives delegated to the Council from each top-level team.
```

**Comment by eholk** [comment]
I checked my box. Please consider that durable, and feel free to merge subsequent edits without an explicit signoff from me.

**Comment by khionu** [comment]
I second Mara's suggestion

File: README.md (hunk 2 above)
Comment (rylev) [reply README.md:9]: This is a bit complicated is its meant to be a direct quote of the RFC, but I will change it to use brackets to indicate a change from the direct quote: "[is]"

**Comment by rylev** [comment]
I agree with @m-ou-se's comment, but eventually I'd like for us to establish clearer procedures for how to interact with the Council depending on what the person in question is trying to achieve. While it's great for us to have a way to "come by and chat", this might not always be the best course of action. For example, I imagine we'll have a way to request the Council to make a decision on something, and I think it would best to document the best way for folks to do this that isn't "just use Zulip". 

All that is to say, I'll add the Zulip link, and then separately do some thinking about additional communication channels.

**Comment by rylev** [comment]
I pushed a commit that tries to address all the feedback. Please uncheck your checkbox if you're unhappy with the result (I'm treating the checkboxes as durable since the changes are small and all of this is very easy to reverse if something undesirable sneaks in). 

Once we get ✅ from the remaining reps, I'll merge.

**Comment by rylev** [comment]
Merging with all 9 ✅

**Pull request merged**